src/state/learnocaml_store.ml

```diff
@@ -335,7 +323,7 @@ module Token = struct
       | Unix.Unix_error (Unix.EEXIST, _, _) -> aux ()
       | e -> Lwt.fail e
     in
-    aux ()
+    aux () >>= fun t -> TokenIndex.add_token !sync_dir t >|= fun _ -> t
```

Comment:
If the index doesn't exist yet, `add_token` will scan, including the just created token, then add it again, resulting in a duplicate.